jax/interpreters/partial_eval.py

```diff
@@ -1169,6 +1169,9 @@ def post_process_custom_vjp_call(self, out_tracers, params):
     assert False  # unreachable
 
 def _memoize(thunk):
+  if config.jax_check_tracer_leaks:
```

Comment:
Actually, I'm concerned about this change. It seems like it might lead to infinite recursions when leak checking is on. Is it necessary for this PR?

Reply:
Hrm or maybe this memoization only has to do with efficiency, not whether there are infinite recursions. I forget! I'll try a test...

Reply:
Yeah, basically my concern is that I think `python tests/api_test.py CustomJVPTest` no longer passes when we make this function return `thunk`.

Reply:
I think this is a repro, tested against this branch:

```python
from jax import custom_jvp, jvp, lax, check_tracer_leaks

@custom_jvp
def f(x):
  return x

@f.defjvp
def f_jvp(primals, tangents):
  (x,), (xdot,) = primals, tangents
  return f(x), xdot

with check_tracer_leaks(True):
  jvp(lambda x: lax.cond(True, f, f, x), (3.,), (1.,))
```

But if we delete these two lines, the above repro still works and the new test you added passes too.

Reply:
sorry, this was a left-over from me trying to fix the actual false positive leak in `custom_jvp` (described here: https://github.com/google/jax/issues/5636#issuecomment-774290760). I gave up on fixing that one (couldn't figure out how to remove the main trace from the `jvp_jaxpr_thunk` closure), and I thought it couldn't hurt to include.
I'll remove it!